piwarsimulator.py

```diff
@@ -166,50 +185,64 @@ def sigint_handler(signal_received, frame):
+        from controllers.SimulatedGripperController import SimulatedGripperController
+        from sensors.SimulatedLineOfSight import SimulatedLineOfSight
+        from sensors.SimulatedBeamSensor import SimulatedBeamSensor
+
+        if not util.is_true(args.simplevision):
+            from sensors.SimulatedVision360 import SimulatedVision360 as SimulatedVision
+        else:
+            from sensors.SimulatedReducedVision import (
+                SimulatedReducedVision as SimulatedVision,
+            )
+    case OperationMode.SensorSimulation:
+        # real control hardware, fake sensors
+        util.import_serial()
+        from controllers.SimulatedMovementController import SimulatedMovementController
+        from controllers.SimulatedGripperController import SimulatedGripperController
+        from controllers.MovementController import MovementController
+        from controllers.GripperController import GripperController
+        from sensors.SimulatedLineOfSight import SimulatedLineOfSight
+        from sensors.SimulatedBeamSensor import SimulatedBeamSensor
+
+        if not util.is_true(args.simplevision):
+            from sensors.SimulatedVision360 import SimulatedVision360 as SimulatedVision
+        else:
+            from sensors.SimulatedReducedVision import (
+                SimulatedReducedVision as SimulatedVision,
+            )
+    case OperationMode.Control:
+        # real control hardware, real sensors
+        util.import_serial()
+        from controllers.MovementController import MovementController
+        from controllers.GripperController import GripperController
+        from controllers.LauncherController import LauncherController
+        from sensors.DistanceSensor import DistanceSensor
+        from sensors.Vision360 import Vision360
+
+    case OperationMode.ControlSimulation:
+        # TODO fake control hardware, real sensors
+        util.import_serial()
+        from controllers.SimulatedMovementController import SimulatedMovementController
+        from controllers.SimulatedGripperController import SimulatedGripperController
+        from sensors.DistanceSensor import DistanceSensor
+        # TODO: Probably needs beam sensor here
+        from sensors.Vision360 import Vision360
+
+    case OperationMode.EverythingSimButVision:
+        # Basically "control_simulation, but with everything in sim except the 360 camera
+        from controllers.SimulatedMovementController import SimulatedMovementController
+        from controllers.SimulatedGripperController import SimulatedGripperController
+        from sensors.SimulatedLineOfSight import SimulatedLineOfSight
+        from sensors.SimulatedBeamSensor import SimulatedBeamSensor
+        from sensors.Vision360 import Vision360
 
 # do serial stuff if needed
-if not args.mode.lower() == "simulation":
+if not (args in [OperationMode.Simulation, OperationMode.EverythingSimButVision]):
```

Comment:
I would think this should be `args.mode in`? (It's that way in the next code block...)

Reply:
That's correct. I'll change it.